File: project_euler/Problem001/Problem001.py
class Problem001:
    def solution(self) -> None:
        target = 1000
        result = self.__get_summation(target, 3) + self.__get_summation(target, 5) - self.__get_summation(target, 3 * 5)
        print(result)

    def __get_summation(self, border: int, factor: int) -> int:
        count = (border - 1) // factor
        return factor * count * (count + 1) >> 1

File: project_euler/Problem001/Problem001.java
package project_euler;

public class Problem001 {
    public void solution() {
        int target = 1000;
        int result = getSummation(target, 3) + getSummation(target, 5) - getSummation(target, 3 * 5);
        System.out.println(result);
    }

    private int getSummation(int border, int factor) {
        int count = (border - 1) / factor;
        return factor * count * (count + 1) >> 1;
    }
}
